test/SubjectServices_test.cpp

```diff
@@ -82,7 +82,8 @@ TEST(SubjectServices, displayOneSubject) {
 
         string output = testing::internal::GetCapturedStderr();
 
-        EXPECT_EQ(output,"The subject ID : 3 not found\n");
+        EXPECT_TRUE(output.find("The subject ID : 3 not found") != string::npos);
+
```

Comment:
Forget to replace directly string comparison assert by sting.find function.

Need to pay attention to avoid using 'directly string comparison assert '.

This test can be performed on local testing environment but it can not pass online TRAVIS CI testing due to different database_condition configuration set.

https://travis-ci.com/github/UAws/OOP-Project/jobs/392415270#L197

`sed -i 's/false/true/g' utils/database_connection.cpp`

https://travis-ci.com/github/UAws/OOP-Project/jobs/392415270#L1054
```
[ RUN      ] SubjectServices.displaySubjectByName
MySQL debug: Executing: 'SELECT subject.subject_id,subject.name FROM subject WHERE subject.name LIKE('%subject%')'
MySQL debug: Constructing result, using handle at 0x55acdbcf9da0
MySQL debug: Accessing next row of handle at 0x55acdbcf9da0
MySQL debug: Accessing next row of handle at 0x55acdbcf9da0
MySQL debug: Accessing next row of handle at 0x55acdbcf9da0
/home/travis/build/UAws/OOP-Project/test/SubjectServices_test.cpp:113: Failure
Expected equality of these values:
  output
    Which is: "MySQL debug: Executing: 'SELECT subject.subject_id,subject.name FROM subject WHERE subject.name LIKE('%IcanNotBeFound%')'\nMySQL debug: Constructing result, using handle at 0x55acdbcea9f0\nMySQL debug: Accessing next row of handle at 0x55acdbcea9f0\nThe subject name : IcanNotBeFound not found\n"
  "The subject name : IcanNotBeFound not found\n"
With diff:
@@ -1,4 @@
-MySQL debug: Executing: 'SELECT subject.subject_id,subject.name FROM subject WHERE subject.name LIKE('%IcanNotBeFound%')'
-MySQL debug: Constructing result, using handle at 0x55acdbcea9f0
-MySQL debug: Accessing next row of handle at 0x55acdbcea9f0
 The subject name : IcanNotBeFound not found\n
[  FAILED  ] SubjectServices.displaySubjectByName (7 ms)
```